Scalar.Common/Git/GitProcess.cs

```diff
@@ -604,6 +604,65 @@ public Result RemoteAdd(string remoteName, string url)
             return this.InvokeGitAgainstDotGitFolder("remote add " + remoteName + " " + url);
         }
 
+        public bool TryGetRemoteDefaultBranch(string remoteName, out string defaultBranch, out string error)
+        {
+            defaultBranch = null;
+            error = null;
+
+            Result lsRemoteResult = this.InvokeGitAgainstDotGitFolder("ls-remote --symref origin HEAD");
+            if (lsRemoteResult.ExitCodeIsFailure)
+            {
+                error = "Failed to call ls-remote to determine remote HEAD";
+                return false;
+            }
+
+            string output = lsRemoteResult.Output;
+
+            const string refPrefix = "ref: ";
+            const string headsPrefix = "ref: refs/heads/";
+            const string suffix = "\tHEAD";
+
+            var cmp = StringComparison.Ordinal;
+            foreach (string line in output.Split('\n'))
+            {
+                int suffixStart = line.IndexOf(suffix, cmp);
+                if (line.StartsWith(refPrefix, cmp) && suffixStart > -1)
+                {
+                    // HEAD is a branch
+                    int prefixStart = line.IndexOf(headsPrefix, cmp);
+                    int prefixEnd = prefixStart + headsPrefix.Length;
+                    if (prefixStart > -1 && prefixEnd < suffixStart)
```

Comment:
Shouldn't this be `if (line.StartsWith(headsPrefix, cmp) && prefixEnd > 0)`? We do not really need to verify that the `headsPrefix` and the `suffix` don't overlap: they cannot (they do not even have a single character in common).

Reply:
I'm trying to avoid scanning over the string yet again here, since the `prefixEnd` comes via a call to `line.IndexOf`, and doing a `line.StartsWith` would be another scan.

Reply:
Actually.. this can be simplified further.. (please wait) ⏳